streams-contrib/streams-processor-regex/src/main/java/org/apache/streams/regex/RegexUtils.java

```diff
@@ -36,6 +36,7 @@
 
     private static final Map<String, Pattern> patternCache = Maps.newConcurrentMap();
     private final static Logger LOGGER = LoggerFactory.getLogger(RegexUtils.class);
+    private final static long REGEX_MATCH_THREAD_KEEP_ALIVE_DEFAULT_TIMEOUT = 10000L; //10 seconds
```

Comment:
Fair enough, I figured I'd give it a very generous amount of time to complete, but I can lower it down to a second if that will be sufficient.